### ISSUE TITLE
Wrong Codec type when updating job

### ISSUE THREAD

**Issue by erictune** [kubernetes/kubernetes#14682]
Symptoms:
```console
$ kubectl create -f docs/user-guide/job.yaml
job "pi" created
# Wait for it to start running (not sure if this is strictly necessary)
$ kubectl delete -f docs/user-guide/job.yaml
```
The latter step hangs forever.
So far, in debugging, I have found that:

- kubectl is waiting in the reaper loop
- it is trying to scale down the Job resource.
- it has successfully GETted it, and is trying to PUT (Update) it.
- it gets this error: `Scaling the controller failed with: no kind "Job" is registered for version "v1"; Current resource version 728999`
- this error is coming from the client, not the server
- the client is doing conversion in preparation for sending the Update, and it gets this error.
- when I print out the RESTClient.Codec, it has version "v1".  This is not right for an experimental Kind like Job.

@caesarxuchao @lavalamp helped me get this far.

**Comment by derekwaynecarr** [comment]
I hit this this morning but didn't have time to debug as far.  Thanks for
the write up.

**Comment by erictune** [comment]
should be fixed by #14683
